gammapy/utils/fitting/datasets.py

```diff
@@ -57,11 +58,13 @@ def __init__(self, datasets, mask=None):
 
         self.mask = mask
 
+    @lazyproperty
```

Comment:
Is it safe to make `Datasets.parameters` a `lazyproperty`? 

Imagine you want to test another spectral model by doing something like this:
```
for ds in datasets:
      ds.model = new_model
``` 
Will the `datasets.parameters` be update accordingly?

Reply:
Good point. This is a problem that arises when we set / modify the models on datasets after init. I'll check again what makes sense here and then change it consistently on all dataset classes.

Reply:
I've looked into this a bit and currently there is the problem, that we attach the covariance matrix to the `Parameters` object. If I declare `Datasets.parameters` as a normal property I can't set the the covariance anymore, because anytime the `.parameters` attribute  is accessed a new merged `Parameters` object is created from the sub-datasets and the covariance info is lost (this is the error we see on Travis-CI). 

For the parallel evaluation of datasets we have to re-think the parameter handling and distribution of parameters in the `Datasets` class anyway. So for now I'll probably leave it as is  with a `lazyproperty`, but we have to come up with a better solution and documentation, how to set a global model for `Datasets` or how to handle the covariance.